app/assets/javascripts/school_administrator_dashboard/page_container.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import _ from 'lodash';
+import SchoolAbsenceDashboard from './school_absence_dashboard.jsx';
+
+export default React.createClass({
+  displayName: 'DashboardPageContainer',
+
+  propTypes: {
+    attendanceData: React.PropTypes.object.isRequired  //should rename more generically
+  },
+
+  //necessary?
+  // studentAbsences: function () {
+  //   const students = this.props.attendanceData;
+
+  // },
+
+  //returns map of homerooms to absence events
+  absencesByHomeroom: function() {
+    //Group data by homeroom
+    const homeRoomGroups = _.groupBy(this.props.attendanceData.absences, 'homeroom');
+    let homeroomAbsences = {};
+    //Simplify homeroom data into map of homerooms to total homeroom absences along with homeroom size
+    Object.keys(homeRoomGroups).forEach( (key) => {
+      const size = homeRoomGroups[key].length;
+      let key2 = key;
+      if (key == "null") key2 = "No Homeroom";
```

Comment:
@edavidsonsawyer Is this the same as `(key == null) ? "No Homeroom" : key`?

Reply:
I don't think so, b/c I have to use the original key value to access the object later on.